clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithArtifactsSpec.java

```diff
@@ -0,0 +1,300 @@
+
+import static io.restassured.RestAssured.get;
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+
+import com.amazonaws.services.applicationautoscaling.AWSApplicationAutoScalingClient;
+import com.amazonaws.services.applicationautoscaling.model.DescribeScalableTargetsRequest;
+import com.amazonaws.services.applicationautoscaling.model.DescribeScalableTargetsResult;
+import com.amazonaws.services.ecs.AmazonECS;
+import com.amazonaws.services.ecs.model.*;
+import com.amazonaws.services.elasticloadbalancingv2.AmazonElasticLoadBalancing;
+import com.amazonaws.services.elasticloadbalancingv2.model.DescribeTargetGroupsRequest;
+import com.amazonaws.services.elasticloadbalancingv2.model.DescribeTargetGroupsResult;
+import com.amazonaws.services.elasticloadbalancingv2.model.TargetGroup;
+import com.netflix.spinnaker.clouddriver.artifacts.ArtifactCredentialsRepository;
+import com.netflix.spinnaker.clouddriver.artifacts.ArtifactDownloader;
+import com.netflix.spinnaker.clouddriver.artifacts.config.ArtifactCredentials;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.clouddriver.aws.security.NetflixAmazonCredentials;
+import com.netflix.spinnaker.clouddriver.ecs.EcsSpec;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsRepository;
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import io.restassured.http.ContentType;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.stubbing.Answer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+public class CreateServerGroupWithArtifactsSpec extends EcsSpec {
+
+  @Autowired AccountCredentialsRepository accountCredentialsRepository;
+
+  @Autowired protected AmazonClientProvider mockAwsProvider;
+
+  @MockBean ArtifactDownloader artifactDownloader;
+
+  @MockBean AWSApplicationAutoScalingClient mockAWSApplicationAutoScalingClient;
+
+  @MockBean ArtifactCredentialsRepository mockArtifactCredentialsRepository;
+
+  @MockBean ArtifactCredentials mockArtifactCredentials;
+
+  private AmazonECS mockECS = mock(AmazonECS.class);
+
+  private AmazonElasticLoadBalancing mockELB = mock(AmazonElasticLoadBalancing.class);
+
+  private final int TASK_RETRY_SECONDS = 3;
+
+  private static final String CREATE_SG_TEST_PATH = "/ecs/ops/createServerGroup";
+
+  @BeforeEach
+  public void setup() {
+
+    // mocking calls
+    when(mockAwsProvider.getAmazonEcs(
```

Comment:
we're mocking calls to several services/beans here, but they're not grouped together. Can we organize this func so that all the calls to a specific mock (e.g., `mockECS`) are grouped together, with calls to `mockAwsProvider` at the end? (since this is the mock bean which is returning all the other AWS service mocks). that way it's easier to see everything a specific service/mock needs to do, and its organized more topologically.